pipeline-scripts/buildlib.groovy

```diff
@@ -173,16 +161,27 @@ def setup_venv() {
             throw err
         }
     }
-
     commonlib.shell(script: "rm -rf art-tools;  git clone -b ${branch_name} https://github.com/${owner}/art-tools.git")
-    commonlib.shell(script: "cd art-tools && ./install.sh")
 
-    out = sh(
-        script: 'pip list | grep "doozer\\|elliott\\|pyartcd"',
-        returnStdout: true
-    )
-    echo "Installed art-tools:"
-    echo "${out}"
+    // Preparing venv for ART tools
+    // The following commands will run automatically every time one of our jobs
+    // loads buildlib (ideally, once per pipeline)
+    withEnv(['UV_LINK_MODE=symlink', 'PATH+MYCARGO=~/.cargo/bin']) {
+        VIRTUAL_ENV = "${env.WORKSPACE}/art-venv"
+        commonlib.shell(script: """
+            rm -rf ${VIRTUAL_ENV}
+            uv venv --python 3.11 --system-site-packages ${VIRTUAL_ENV}
+            source ${VIRTUAL_ENV}/bin/activate
+            uv pip install --upgrade pip setuptools
```

Comment:
curious if we can use --seed arg for uv venv

Reply:
Looks like installing setuptools is not needed. I have removed this line from the script